core/src/components/datetime/datetime-util.ts

```diff
@@ -324,6 +324,16 @@ export function updateDate(existingData: DatetimeData, newData: any): boolean {
         (existingData as any)[key] = newData[key].value;
       }
       return true;
+    } else if (newData.ampm) {
+      newData.hour = {
```

Comment:
Are we able to do `newData.hour.value` here to get rid of an extra layer?

Reply:
I tried but couldn't see a nice way to do it because when users change AM/PM `newData.hour` is undefined. I'll take another look.

Reply:
Ahhh good point ok never mind then 👍

Reply:
Actually, I replaced one of the ternary ifs and I think it's more readable now.